File: wrappers/wasm/test/did/did_key.test.ts
import { DIDKey, InMemKms } from "../../pkg";
import { KeyType } from "../../types";

describe("did:key: ", () => {
  test("generate DID", async () => {
    const kms = new InMemKms();
    const kid = await kms.create(KeyType.P256);
    const keyHandle = await kms.get(kid);

    const didKey = new DIDKey();

    let did = didKey.generate(keyHandle);

    expect(did.startsWith("did:key")).toBeTruthy();
  });
});

File: wrappers/uniffi/swift/Tests/EqusSdkTests/DIDKeyTest.swift
import Testing
import Foundation
import Swifter
@testable import EqusSdk

@Suite(.serialized) class DIDKeyTests {

	@Test func generateDID() async throws {
		let kms = InMemKms()
		let kid = try await kms.create(kt: KeyType.p256)
		let keyHandle = try await kms.get(kid: kid)

		let didKey = DidKey()

		let did = try didKey.generate(key: keyHandle)

		#expect(did.contains("did:key:"))
	}
}